### PR TITLE
Support multi arch image

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 FROM quay.io/bitnami/golang:1.17 AS builder
 WORKDIR /go/src/open-cluster-management.io/registration
 COPY . .
 ENV GO_PACKAGE open-cluster-management.io/registration
 
-RUN make build --warn-undefined-variables
-RUN make build-e2e --warn-undefined-variables
+RUN GOOS=${OS} GOARCH=${ARCH} make build --warn-undefined-variables
 
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```
Comment (yue9944882) [inline Dockerfile:12]: replace this w/ a multi-arch base image?

File: Makefile [hunk 2]
```diff
@@ -32,7 +31,7 @@ kustomize_dir:=$(dir $(KUSTOMIZE))
 # $2 - Dockerfile path
 # $3 - context directory for image build
 # It will generate target "image-$(1)" for building the image and binding it as a prerequisite to target "images".
-$(call build-image,registration,$(IMAGE_REGISTRY)/registration,./Dockerfile,.)
+$(call build-image,registration,$(IMAGE_REGISTRY)/registration:$(IMAGE_TAG),./Dockerfile,.)
```
Comment (yue9944882) [inline Makefile:32]: how is `OS` and `ARCH` overrided upon building the image

File: .github/workflows/go-postsubmit.yml [hunk 3]
```diff
@@ -31,9 +34,34 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
       - name: install imagebuilder
         run: go install github.com/openshift/imagebuilder/cmd/imagebuilder@v1.2.1
+      - name: pull base image
+        run: docker pull registry.access.redhat.com/ubi8/ubi-minimal:latest --platform=linux/${{ matrix.arch }}
       - name: images
-        run: make images
+        run: |
+          IMAGE_TAG=latest-${{ matrix.arch }} \
+          IMAGE_BUILD_EXTRA_FLAGS="--build-arg OS=linux --build-arg ARCH=${{ matrix.arch }}" \
```
Comment (yue9944882) [inline .github/workflows/go-postsubmit.yml:42]: ah i see

File: Dockerfile [hunk 4]
```diff
@@ -1,16 +1,15 @@
+ARG OS=linux
+ARG ARCH=amd64
 FROM quay.io/bitnami/golang:1.17 AS builder
```
Comment (yue9944882) [inline Dockerfile]: repeat below:

```
ARG OS
ARG ARCH
```

File: Dockerfile (hunk 1 above)
Comment (qiujian16) [reply Dockerfile:12]: I have to pull base image at first, since there is no separate tag for different platform

**Review by yue9944882 — COMMENTED** [review]
/lgtm

**Pull request merged**